Update code with Devel-PPPort recommendations

atoomic opened the pull request:
view CI build run output from https://github.com/atoomic/version.pm/actions/runs/199840375

Leont commented:
This doesn't quite look right to me.

Edit: now that I've run the code myself, I'm pretty sure it's wrong. For any other dist it would make sense, we we provide these functions.

atoomic commented on vutil/vutil.h:
```diff
@@ -2,6 +2,10 @@
+#define NEED_vnewSVpvf
+#define NEED_my_snprintf
+#define NEED_newRV_noinc
+#define NEED_sv_2pv_flags_GLOBAL
```
this is a change we need... going to open a new PR for it..

atoomic commented on vutil/vxs.xs:
```diff
@@ -4,13 +4,7 @@
-#define NEED_newSVpvn_flags_GLOBAL
-#define NEED_warner
 /* end vutil.c */
-#include "ppport.h"
```
this is not needed as ppport.h is going to be added from vutil.h

Leont replied:
First of all, that really ought to be a separate commit.

Second of all, why would you want that in the first place? What's the advantage of moving it into vutil.h? I would make more sense to me to eliminate the `#include "ppport.h"` there, given that it has to be guarded in an `#ifndef PERL_CORE`

atoomic replied:
#19 is splitting the `ppport.h` setup as its own PR
let's move the discussion there

atoomic commented:
I would have preferred not close this PR... going to open another one with just the ppport.h change and `NEED_*`

Leont commented:
> I would have preferred not close this PR... going to open another one with just the ppport.h change and NEED_*

Are you sure you set the right `--compat-version=X`. The default is like `5.003`, it didn't look like I'd need any `NEED_` if I set it to `5.6.1` (if we did, we wouldn't be passing tests on older perls)

atoomic committed:
Apply Devel-PPPort recomendations

Also note that the NEED_* need to be set before
the very first load of ppport.h

perl vutil/ppport.h --compat-version=5.6.1

atoomic commented:
@Leont you can view b3c802a0 with recommendations using 
`perl vutil/ppport.h --compat-version=5.6.1`

Also note that some of the `NEED_*` were set too late, as this need to occur before the first time `ppport.h` is loaded

you can view a test run from: https://github.com/atoomic/version.pm/actions/runs/204670943

Leont commented on vutil/vutil.h:
```diff
@@ -151,33 +157,33 @@ static const char * Perl_prescan_version2(pTHX_ const char *s, bool strict, cons
 
 #else
 
-const char * Perl_scan_version(pTHX_ const char *s, SV *rv, bool qv);
```
This is plain wrong.

It would be silly for ordinary dists, but it's exactly right for core, and given that this is shared with core it should remain like it is.
